File: JavaTest/src/learn/j18/demo/Persion.java
package learn.j18.demo;

/**
 * Created by better on 2017/6/14.
 */
public class Persion {
    private String name;
    private int age;
    private double money;

    public Persion(String name, int age, double money) {
        this.name = name;
        this.age = age;
        this.money = money;
    }

    @Override
    public String toString() {
        return name + ", " + age +
                ", " + money;
    }

    public int getAge() {
        return age;
    }

    public void setAge(int age) {
        this.age = age;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public double getMoney() {
        return money;
    }

    public void setMoney(double money) {
        this.money = money;
    }
}

File: HelloKotlin/src/better/kotlin/learn/objects/Persion.kt
package better.kotlin.learn.objects

/**
 * 主构造函数，相当于申明了类的属性
 * Created by better on 2017/5/30.
 */
open class Persion constructor(var name: String, var sex: Int, age: Int) {
    /**
     * 申明可变属性，还必须要初始化,否则就abstract
     */
    var age: Int = age
        set(value) {
            age = value
        }
    /**
     * 申明不可变属性,在构造函数中初始化
     */
    val id: Int?

    init {
        id = 10
    }

    /**
     * 申明不可变属性
     */
    val id2 = 10
    /**
     * 只能自己看到
     */
    private var tag: Long = 10

    constructor(name: String, sex: Int, age: Int, tag: String) : this(name, sex, age)

    constructor(tag: String) : this("default", 1, 0)

    override fun toString(): String {
        return "name:$name,sex:$sex"
    }
}